src/main/java/hudson/plugins/performance/PerformancePublisher.java

```diff
@@ -278,499 +328,542 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListen
     double thresholdTolerance = 0.00000001;
     Result result = Result.SUCCESS;
     EnvVars env = build.getEnvironment(listener);
+    
+    Collection<PerformanceReport> parsedReports = null;
+    String glob = null;
+    
+    /*
+     * preparing evaluation - this is necessary regardless of the mode of evaluation
+     */
+    
+    // add the report to the build object.
+    PerformanceBuildAction a = new PerformanceBuildAction(build, logger, parsers);
```

Comment:
This was not really my change. Just changed the structure of the  perform() method because of the "Evaluation Mode" I introduced.

Before:
perform(...){
// preparation code
// standard evaluation code
}

Now:
perform(...){
// preparation code
if(modeEvaluation){
   // standard evaluation code
} else {
   // expert evaluation code
}

Due to all tests run green, I am confident, that this will work.